perf-tool/src/client.cpp

```diff
@@ -166,16 +190,29 @@ void Client::closeClient()
 int main(int argc, char const *argv[])
 {
     string hostname = "localhost";
+    std::string clientLogPath = "clientLogs.json";
     int portno = ServerConstants::DEFAULT_PORT;
 
     if (argc > 1)
     {
         portno = atoi(argv[1]);
         if (argc > 2)
-            hostname = argv[2];
+        {
+            if (strcmp(argv[2], "--log") == 0)
+                clientLogPath = argv[3];
+            else if (strcmp(argv[2], "--hostname") == 0)
+                hostname = argv[3];
+            if (argc > 4)
+            {
+                if (strcmp(argv[4], "--log") == 0)
+                     clientLogPath = argv[5];
+                else if (strcmp(argv[4], "--hostname") == 0)
+                    hostname = argv[5];
+            }
+        }
```

Comment:
This solution is not very robust in the sense that it assume that if `argv[2]` exists, then `argv[3]` also exists.
How about using `getopt_long` https://linux.die.net/man/3/getopt_long

Reply:
@marius, I tried implementing with getopt_long. But still there is one issue with this new commit. If I run as  `./a.out 9002 --log --host localhost` then log takes input as --host which is incorrect.

Any suggestions on how I can handle this scenario would be helpful?

Reply:
Based on the description it should have worked.
An workaround is to make the option value optional (not required) and then complain if `optarg` is NULL.
BTW, maybe it makes sense to have an option for port as well. If port is not specified, complain and exit.